x-pack/libbeat/common/cloudfoundry/events.go

```diff
@@ -0,0 +1,525 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cloudfoundry
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+
+	"github.com/cloudfoundry/sonde-go/events"
+)
+
+// EventType defines the different event types that can be raised from RPLClient.
+type EventType uint
+
+const (
+	// EventTypeHttpAccess is a http access event.
+	EventTypeHttpAccess EventType = iota
+	// EventTypeLog is a log event.
+	EventTypeLog
+	// EventTypeCounter is a counter event.
+	EventTypeCounter
+	// EventTypeValueMetric is a value metric event.
+	EventTypeValueMetric
+	// EventTypeContainerMetric is a container metric event.
+	EventTypeContainerMetric
+	// EventTypeError is an error event.
+	EventTypeError
+)
+
+// String returns string representation of the event type.
+func (t EventType) String() string {
+	switch t {
+	case EventTypeHttpAccess:
```

Comment:
Is there any place where these are documented? It would be nice to link it in a comment

Reply:
I added the link to the documentation over the const.